### DISCUSSION TITLE
express server, useRouter and getServerSideProps not working

### DISCUSSION THREAD

**Discussion by marcossouzadotdev** [vercel/next.js#11813] [Help]
## Problem 1 with express:
I have this in my `server.js` file:

```
// Article Page
server.get("/:slug", (request, response) => {
	console.log(request);
	return app.render(request, response, "/[slug]", request.params);
});
```

But when i access a link like: `http://localhost:3001/meeeeeee`
The `request.params` returns this:  `params: { slug: 'favicon.ico' },`

##  Problem 2 with `getServerSideProps`:

I added a `console.log` in the context inserted by next just to test:

```
export const getServerSideProps = async (context) => {
	console.log("========= CONTEXT ", context);
	return {
		props: { context },
	};
};

```
But this does not display any logs, i.e., it is not running, I think. I also try with the doc's code using a normal `function`.

##  Problem 3 with the `useRouter`:

I cant access the `router.query`, this return this object: `{ amp: undefined }`
```
const Article = (props) => {
	const router = useRouter();
	console.log(router.query);
	return <h1>Hello</h1>;
};

// output
============ PROPS {}
===== router  ServerRouter {
  route: '/[slug]',
  pathname: '/[slug]',
  query: { amp: undefined },
  asPath: '/[slug]',
  isFallback: false
}
```
This is displayed two times when i access the url, but in the browser console, at first time returns a empty object, and at two time, this returns the correct slug:

```
[slug].jsx:6  {}
[slug].jsx:6 {slug: "meeeeeee"}
```

Note: I am accessing the `"/[slug]"` route directly in the browser, without generating clickable links for this before.

Complete `[slug].jsx` file with the `Article` component:

```
import React from "react";
import { useRouter } from "next/router";

const Article = (props) => {
	const router = useRouter();
	console.log(router.query);
	console.log(props); // returns empty object every time
	return <h1>Hello</h1>;
};

export async function getServerSideProps(context) {
	// TODO:
	// call the api to get post
	// const slug = context.params.slug;
	console.log("========== context", context);
	return {
	    props: { context, slug },
	};
}

export default Article;
```

Complete server.js file:

```
const express = require("express");
const next = require("next");

const port = parseInt(process.env.PORT, 10) || 3001;

const dev = process.env.NODE_ENV !== "production";

const app = next({ dev });
const handle = app.getRequestHandler();

const expressHandle = () => {
	const server = express();

	// Home Page | Post list
	server.get("/", (request, response) => {
		return app.render(request, response, "/", request.params);
	});

	// Article Page
	server.get("/:slug", (request, response) => {
		return app.render(request, response, "/[slug]", request.params);
	});

	// handle errors like server error
	server.all("*", (req, res) => {
		return handle(req, res);
	});

	// run server

	server.listen(port, (err) => {
		if (err) throw err;
		console.log(`> Ready on http://localhost:${port}`);
	});
};

app.prepare().then(expressHandle);
```

command to run server: node server.js

source with relevant content: https://github.com/flakesrc/next-errors

**Comment by dominik2323** [comment]
I had similar issue as your first problem with favicon.ico. The cause was Firefox's native behavior that makes an request to /favicon.ico if you didn't provide any favicon from your code.

**Comment by alihbk** [comment]
i have the same issue . do you solve that ?
